ui/src/app/pages/admin/workspace/filename-cell.spec.tsx

```diff
@@ -0,0 +1,148 @@
+import '@testing-library/jest-dom';
+
+import * as React from 'react';
+import { mockNavigate } from 'setupTests';
+
+import { FileDetail } from 'generated/fetch';
+
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import {
+  expectButtonElementDisabled,
+  expectButtonElementEnabled,
+} from 'testing/react-test-helpers';
+
+import { FilenameCell } from './filename-cell';
+
+let storageBucketPath: string,
+  notebookFilename: string,
+  notebookFile: FileDetail,
+  accessReason: string,
+  workspaceNamespace: string;
+
+beforeEach(() => {
+  storageBucketPath = 'gs://testbucket';
+  notebookFilename = 'notes.ipynb';
+  notebookFile = {
+    name: notebookFilename,
+    path: `${storageBucketPath}/notebooks/${notebookFilename}`,
+    lastModifiedTime: new Date().valueOf(),
+    sizeInBytes: 1e6, // 1 MB
+  };
+  accessReason = 'no reason, just being nosy';
+  workspaceNamespace = 'some fake value';
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('FilenameCell', () => {
+  it('allows notebook preview', async () => {
+    const user = userEvent.setup();
+
+    const { container } = render(
+      <FilenameCell
+        {...{ accessReason, workspaceNamespace, storageBucketPath }}
+        file={notebookFile}
+      />
+    );
+    expect(container).toBeInTheDocument();
+    expect(screen.getByText(notebookFilename)).toBeInTheDocument();
+
+    const previewButton = screen.getByRole('button');
```

Comment:
alt: could search for "Preview" text instead?

Reply:
I checked [the RTL docs](https://testing-library.com/docs/queries/about#types-of-queries): "Most often, [getByRole] will be used with the name option like so: `getByRole('button', {name: /submit/i})`"

That seems good to me since we refer to it as "the preview button" and not "the preview element" or "the button" when we talk about it, which implies that both pieces of data (preview, button) are needed to identify it.